### PR TITLE
upgrade to hadoop 2.6.0

### PR DESCRIPTION
**Pull request by infynyxx**

### CODE REVIEW

**Comment by nishantmonu51** [comment]
can you also modify the default in the docs - 
http://druid.io/docs/latest/Peons.html
http://druid.io/docs/latest/Tasks.html

File: indexing-hadoop/pom.xml [hunk 1]
```diff
@@ -67,6 +67,10 @@
       <groupId>org.apache.hadoop</groupId>
       <artifactId>hadoop-client</artifactId>
     </dependency>
+      <dependency>
+          <groupId>org.apache.hadoop</groupId>
+          <artifactId>hadoop-aws</artifactId>
```
Comment (fjy) [inline indexing-hadoop/pom.xml]: this is aws hadoop, which many people do not run on, and will probably not work with stock hadoop. I think a proper way to support multiple hadoop versions is worth a discussion in the forums

**Comment by nishantmonu51** [comment]
CLIHadoopIndexer also needs to be updated.

**Comment by infynyxx** [comment]
@nishantmonu51 Updated via https://github.com/infynyxx/druid/commit/a154e024187edd2b3453b6db049a3ba9910071ee

File: indexing-hadoop/pom.xml (hunk 1 above)
Comment (infynyxx) [inline indexing-hadoop/pom.xml]: Agreed. But if Hadoop 2.6.0 is used during runtime, one will have to also include `hadoop-aws`. 

I will post this in Google groups.

**Comment by fjy** [comment]
@infynyxx There are a couple of merge conflicts here I believe.

@xvrl @cheddar @himanshug @gianm @nishantmonu51 I'd like to hear your thoughts about updating Druid's default Hadoop version. We can start a vote here or in the forums.

**Comment by infynyxx** [comment]
@fjy Sounds good to me.

**Comment by drcrallen** [comment]
If you want to be really judicious about it, perhaps an issue on resolving how to do Hadoop version upgrades going forward would be in order, and discussion should go around that.

I'm specifically interested in making sure people are going to have some idea for if/when Druid would adopt Hadoop 3.x whenever it becomes stable.

File: indexing-hadoop/pom.xml (hunk 1 above)
Comment (himanshug) [inline indexing-hadoop/pom.xml]: hadoop-aws shouldn't be necessary after  #1151 unless you're really running it on aws with native s3 fs.

**Comment by himanshug** [comment]
Aside from one comment wrt hadoo-aws, this patch looks fine to me. 

However, users who are relying on druid to bring hadoop jars, who can't upgrade their hadoop cluster to 2.6,  will need to change their configuration so that they continue to get hadoop-2.3 jars. so, this should go in the release notes.

**Comment by infynyxx** [comment]
@fjy resolved conflict

@himanshug removed `hadoop-aws` dependency

File: docs/content/Tasks.md [hunk 2]
```diff
@@ -120,18 +120,20 @@ The Hadoop Index Task is used to index larger data sets that require the paralle
...
-|hadoopCoordinates|The Maven \<groupId\>:\<artifactId\>:\<version\> of Hadoop to use. The default is "org.apache.hadoop:hadoop-client:2.3.0".|no|
+|hadoopCoordinates|The Maven \<groupId\>:\<artifactId\>:\<version\> of Hadoop to use. The default is "org.apache.hadoop:hadoop-client:2.6.0".|no|
 
 
 The Hadoop Index Config submitted as part of an Hadoop Index Task is identical to the Hadoop Index Config used by the `HadoopBatchIndexer` except that three fields must be omitted: `segmentOutputPath`, `workingPath`, `updaterJobSpec`. The Indexing Service takes care of setting these fields internally.
 
 #### Using your own Hadoop distribution
 
-Druid is compiled against Apache hadoop-client 2.3.0. However, if you happen to use a different flavor of hadoop that is API compatible with hadoop-client 2.3.0, you should only have to change the hadoopCoordinates property to point to the maven artifact used by your distribution. For non-API compatible versions, please see [here](Other-Hadoop.html).
+<<<<<<< HEAD
```
Comment (drcrallen) [inline docs/content/Tasks.md]: Conflict tag

Comment (infynyxx) [inline docs/content/Tasks.md]: Fixed via a2d1b05

**Comment by xvrl** [comment]
@infynyxx Unless we have a good reason to require anything older than 2.3, I would prefer we leave the default at the minimum version required, and add separate build profiles for different versions / flavors of hadoop, [similar to what Spark does](https://github.com/apache/spark/blob/master/pom.xml#L1633)

**Comment by infynyxx** [comment]
@xvrl Sounds good. Closing this PR.

**Pull request closed without merging**